polus-training-splinedist-plugin/src/utils.py

```diff
@@ -0,0 +1,520 @@
+    num_images_tested = len(X_val)
+    num_labels_tested = len(Y_val)
+    
+    # Get logs for end user
+    totalimages = num_images_trained+num_images_tested
+    logger.info("{}/{} images used for training".format(num_images_trained, totalimages))
+    logger.info("{}/{} labels used for training".format(num_labels_trained, totalimages))
+    logger.info("{}/{} images used for testing".format(num_images_tested, totalimages))
+    logger.info("{}/{} images used for testing".format(num_labels_tested, totalimages))
+
+
+    # Renamed inputs
+    del input_images
+    del input_labels
+    del num_inputs
+
+    # assertions
+    assert num_images_trained > 1, "Not Enough Training Data"
+    assert num_images_trained == num_labels_trained, "The number of images does not match the number of ground truths for training"
+    assert num_images_tested == num_labels_tested, "The number of images does not match the number of ground for testing"
+    num_trained = num_images_trained
+    num_tested = num_images_tested
+
+    del num_images_trained
+    del num_images_tested
+    del num_labels_trained
+    del num_labels_tested
+
+    # Need a list of numpy arrays to feed to SplineDist
+    array_images_trained = []
+    array_labels_trained = []
+    array_images_tested = []
+    array_labels_tested = []
+
+    # Neural network parameters
+    axis_norm = (0,1)
+    n_channel = 1 # this is based on the input data
+    
+    model_dir_name = '.'
+    model_dir_path = os.path.join(output_directory, model_dir_name)
+
+    # Get the testing data for neural network
+    logger.info("\n Starting to Load Test Data ...")
+    tenpercent_tested = np.floor(num_tested/10)
+    for i in range(num_tested):
+
+        image = X_val[i]
+        label = Y_val[i]
+        base_image = os.path.basename(str(image))
+        base_label = os.path.basename(str(label))
+        assert base_image == base_label, "{} and {} do not match".format(base_image, base_label)
+
+        # The original image
+        br_image = BioReader(image)
+        im_array = br_image[:,:,0:1,0:1,0:1]
+        im_array = im_array.reshape(br_image.shape[:2])
+        array_images_tested.append(normalize(im_array,pmin=1,pmax=99.8,axis=axis_norm))
+
+        # The corresponding label for the image
+        br_label = BioReader(label)
+        lab_array = br_label[:,:,0:1,0:1,0:1]
+        lab_array = lab_array.reshape(br_label.shape[:2])
+        array_labels_tested.append(fill_label_holes(lab_array))
+        
+        assert im_array.shape == lab_array.shape, "{} and {} do not have matching shapes".format(base_image, base_label)
+
+        if (i%tenpercent_tested == 0) and (i!=0):
+            logger.info("Loaded ~{}% of Test Data".format(np.ceil((i/num_tested)*100), num_tested))
+    logger.info("Done Loading Testing Data")
+
+    # Read the input images used for training
+    logger.info("\n Starting to Load Train Data ...")
+    tenpercent_trained = np.floor(num_trained/10)
```

Comment:
You can also do integer division to achieve the same result. You don't have to change that here, just something to keep in mind so you know you don't need numpy to perform this task:

`tenpercent_trained = num_trained//10` -> The `//` does floor division.